aptos-move/parallel-executor/src/proptest_types/types.rs

```diff
@@ -29,76 +27,88 @@ use std::{
 #[proptest(params = "TransactionGenParams")]
 pub struct TransactionGen<V: Arbitrary + Debug + 'static + Clone> {
     #[proptest(
-        strategy = "vec((any::<Index>(), value_strategy(params.write_keep_rate)), 1..params.possible_write_size)"
+        strategy = "vec(vec((any::<Index>(), any::<V>()), 1..params.possible_write_size), 1..params.read_write_alternatives)"
+    )]
+    keys_modified: Vec<Vec<(Index, V)>>,
+    #[proptest(
+        strategy = "vec(vec(any::<Index>(), 1..params.read_size), 1..params.read_write_alternatives)"
     )]
-    keys_modified: Vec<(Index, Option<V>)>,
-    #[proptest(strategy = "vec(any::<Index>(), 1..params.read_size)")]
-    keys_read: Vec<Index>,
+    keys_read: Vec<Vec<Index>>,
 }
 
 #[derive(Clone, Copy)]
 pub struct TransactionGenParams {
     pub possible_write_size: usize,
     pub read_size: usize,
-    pub write_keep_rate: f64,
+    pub read_write_alternatives: usize,
 }
 
 /// A naive transaction that could be used to test the correctness and throughput of the system.
 #[derive(Debug, Clone)]
 pub enum Transaction<K, V> {
     Write {
+        /// Incarnation to simulate secondary indices, i.e. incarnations will differ in reads and writes.
+        incarnation: Arc<AtomicUsize>,
         /// Write to some keys with value provided.
-        actual_writes: Vec<(K, V)>,
-        /// Skipp writing to some keys. This is used to simulate over approximation of the inferencer.
-        skipped_writes: Vec<K>,
+        writes: Vec<Vec<(K, V)>>,
         /// Read from some keys.
-        reads: Vec<K>,
+        reads: Vec<Vec<K>>,
```

Comment:
I assume this is incarnation reads? we should add comments about why it's vec of vec now

Reply:
added comments.